src/components/ShoppingList/ShoppingList.js

```diff
@@ -11,13 +13,41 @@ function ShoppingList({ listId }) {
     db.collection(`lists/${listId}/items`).orderBy('purchaseInterval', 'asc'),
   );
 
-  const handleCheck = (e) => {
-    const itemId = e.target.value;
+  // Helper function to get the latest interval between purchases (expects Luxon date objects)
+  const getLatestInterval = ({ lastPurchaseDate, newPurchaseDate }) => {
```

Comment:
So for my own understanding, why do we need to pass our args into this function with {}? I'm used to doing this when we are destructuring out something from an object directly to save using dot notation in the function body, but that doesn't seem to be what we are doing here. Do we need to do this because both variables represent Luxon date objects?

Reply:
There's no reason it needs to be done this way, and is unrelated to Luxon. This is just a style of handling parameters that I've been encouraged to use when the order of the parameters might not be intuitive, or there are just a lot of them so confusion is easy. (It's also a really common pattern in some of the codebases I work with all the time, which use loosely typed languages like JS and PHP.) But we could just as well write it like this: 
`const getLatestInterval = (lastPurchaseDate, newPurchaseDate) => { ... }`

The one advantage of using a single object parameter and destructuring is that you can then call the function with the "arguments" (properties of one argument really) in any order, and because you've explicitly named them it will still work the same way. For example, as is either of these function calls would return the same output:
```
getLatestInterval({ newPurchaseDate, lastPurchaseDate });
getLatestInterval({ lastPurchaseDate, newPurchaseDate });
```